From e2effe48498f6003b6447e4978e6dae12ff34ea7 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:05 -0700
Subject: [PATCH 1630/1852] v4l: xilinx: sdirxss: Fix warnings for NULL_RETURNS

commit 11403fe1719b07265a28473eea6ba6798849c15f from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the NULL_RETURNS warnings.

The actual warnings are as below -

- __xsdirxss_get_pad_format():
Event return_null: 	Explicitly returning null.

- xsdirxss_set_format():
Event returned_null: 	"__xsdirxss_get_pad_format" returns "NULL".
Event var_assigned: 	Assigning: "__format" = "NULL" return value from
"__xsdirxss_get_pad_format".
Event dereference: 	Dereferencing "__format", which is known to be
"NULL".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-sdirxss.c    | 25 +++++++++++++++----
 1 file changed, 20 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index deb53d2e58a6..fd571e8c77da 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1887,14 +1887,22 @@ __xsdirxss_get_pad_format(struct xsdirxss_state *xsdirxss,
 			  struct v4l2_subdev_pad_config *cfg,
 			  unsigned int pad, u32 which)
 {
+	struct v4l2_mbus_framefmt *format;
+
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_format(&xsdirxss->subdev, cfg, pad);
+		format = v4l2_subdev_get_try_format(&xsdirxss->subdev, cfg,
+						    pad);
+		break;
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
-		return &xsdirxss->format;
+		format = &xsdirxss->format;
+		break;
 	default:
-		return NULL;
+		format = NULL;
+		break;
 	}
+
+	return format;
 }
 
 /**
@@ -1913,14 +1921,19 @@ static int xsdirxss_get_format(struct v4l2_subdev *sd,
 {
 	struct xsdirxss_state *xsdirxss = to_xsdirxssstate(sd);
 	struct xsdirxss_core *core = &xsdirxss->core;
+	struct v4l2_mbus_framefmt *format;
 
 	if (!xsdirxss->vidlocked) {
 		dev_err(core->dev, "Video not locked!\n");
 		return -EINVAL;
 	}
 
-	fmt->format = *__xsdirxss_get_pad_format(xsdirxss, cfg,
-						 fmt->pad, fmt->which);
+	format = __xsdirxss_get_pad_format(xsdirxss, cfg,
+					   fmt->pad, fmt->which);
+	if (!format)
+		return -EINVAL;
+
+	fmt->format = *format;
 
 	dev_dbg(core->dev, "Stream width = %d height = %d Field = %d\n",
 		fmt->format.width, fmt->format.height, fmt->format.field);
@@ -1955,6 +1968,8 @@ static int xsdirxss_set_format(struct v4l2_subdev *sd,
 
 	__format = __xsdirxss_get_pad_format(xsdirxss, cfg,
 					     fmt->pad, fmt->which);
+	if (!__format)
+		return -EINVAL;
 
 	/* Currently reset the code to one fixed in hardware */
 	/* TODO : Add checks for width height */
-- 
2.31.1

